### PR TITLE
Use `Listeners.notify` in more places

### PR DESCRIPTION
**Pull request by basil**
Adapting more callers to the new API introduced in #6000.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/model/listeners/RunListener.java
Comment (jglick) [inline core/src/main/java/hudson/model/listeners/RunListener.java:206]: This repeated bit of logic could I guess be encapsulated in a new `static` helper method here. Maybe not worth the bother.

File: core/src/main/java/hudson/model/listeners/RunListener.java
Comment (jglick) [inline core/src/main/java/hudson/model/listeners/RunListener.java:267]: now unused

**Review by jglick — APPROVED** [review]

File: core/src/main/java/hudson/model/listeners/RunListener.java
Comment (basil) [reply core/src/main/java/hudson/model/listeners/RunListener.java:205]: Could indeed, but I have to draw the line somewhere so I won't bother to fix this pre-existing issue.

**Review by daniel-beck — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by timja — APPROVED** [review]

**Pull request merged**
